NuRadioReco/framework/base_trace.py

```diff
@@ -177,8 +195,29 @@ def add_trace_start_time(self, start_time):
     def get_trace_start_time(self):
         return self._trace_start_time
 
-    def get_frequencies(self):
-        return get_frequencies(self.get_number_of_samples(), self._sampling_rate)
+    def get_frequencies(self, window_or_nsamples=None):
+        """
+        Returns the frequencies of the frequency spectrum.
+
+        Parameters
+        ----------
+        window_or_nsamples: array of bools or int (default: None)
+            If not None, used to determine the number of samples in the time domain used for the frequency spectrum.
+
+        Returns
+        -------
+        frequencies: np.array of floats
+            The frequencies of the frequency spectrum.
+        """
+        if window_or_nsamples is None:
+            nsamples = self.get_number_of_samples()
+        else:
+            if isinstance(window_or_nsamples, (int, float)):
+                nsamples = int(window_or_nsamples)
+            else:
+                nsamples = int(np.sum(window_or_nsamples))
+
+        return get_frequencies(nsamples, self._sampling_rate)
```

Comment:
I would probably have a slight preference for having the same `window` argument as in `get_frequency_spectrum`, for simplicity... but I can live with this.